### PR TITLE
[WIP] Adding mention and PM to hovercards

### PR DESCRIPTION
**Pull request by angerella10**
Pull Request for https://github.com/diaspora/diaspora/issues/6882

hovercard-mention-message-modal

The buttons are fully functioning, showing the corresponding modals.

### CODE REVIEW

File: app/assets/javascripts/app/helpers/modal_helper.js [hunk 1]
```diff
@@ -1,12 +1,34 @@
+
 (function(){
-  app.helpers.showModal = function(id){
+  app.helpers.showModal = function(id, statusMessagePath, title){
```
Comment (diaspora-code-review) [inline app/assets/javascripts/app/helpers/modal_helper.js]: Missing space before opening brace.

File: app/assets/javascripts/app/helpers/modal_helper.js [hunk 2]
```diff
@@ -1,12 +1,34 @@
...
     $(id).modal();
-    var modalBody = $(id).find(".modal-body");
 
-    var url = $(id).attr("href");
+    var $modalTitle = $(id).find(".modal-title");
+
+    if($('showMentionModal').modal()) {
```
Comment (diaspora-code-review) [inline app/assets/javascripts/app/helpers/modal_helper.js]: Expected space(s) after "if".

Comment (diaspora-code-review) [inline app/assets/javascripts/app/helpers/modal_helper.js]: Strings must use doublequote.

File: app/assets/javascripts/app/helpers/modal_helper.js [hunk 3]
```diff
@@ -1,12 +1,34 @@
...
+      if(statusMessagePath){
```
Comment (diaspora-code-review) [inline app/assets/javascripts/app/helpers/modal_helper.js]: Expected space(s) after "if".

Comment (diaspora-code-review) [inline app/assets/javascripts/app/helpers/modal_helper.js]: Missing space before opening brace.

File: app/assets/javascripts/app/helpers/modal_helper.js [hunk 4]
```diff
@@ -1,12 +1,34 @@
...
+        if($modalTitle.data("original-title") == null){
```
Comment (diaspora-code-review) [inline app/assets/javascripts/app/helpers/modal_helper.js]: Expected space(s) after "if".

Comment (diaspora-code-review) [inline app/assets/javascripts/app/helpers/modal_helper.js]: Missing space before opening brace.

File: app/assets/javascripts/app/helpers/modal_helper.js [hunk 5]
```diff
@@ -1,12 +1,34 @@
...
+          $modalTitle.data("original-title", $modalTitle[0].textContent);
+        }
+
+        var modalBody = $(id).find(".modal-body");
+
+        $modalTitle.text(title);
+        modalBody.load(statusMessagePath, function(){
```
Comment (diaspora-code-review) [inline app/assets/javascripts/app/helpers/modal_helper.js]: Missing space before opening brace.

File: app/assets/javascripts/app/helpers/modal_helper.js [hunk 6]
```diff
@@ -1,12 +1,34 @@
...
+          $(id).find("#modalWaiter").remove();
+        });
+      }
+      else{
```
Comment (diaspora-code-review) [inline app/assets/javascripts/app/helpers/modal_helper.js]: Expected space(s) after "else".

Comment (diaspora-code-review) [inline app/assets/javascripts/app/helpers/modal_helper.js]: Closing curly brace does not appear on the same line as the subsequent block.

File: app/assets/javascripts/app/helpers/modal_helper.js [hunk 7]
```diff
@@ -1,12 +1,34 @@
...
+      	if($modalTitle.data("original-title") != null){
```
Comment (diaspora-code-review) [inline app/assets/javascripts/app/helpers/modal_helper.js]: Mixed spaces and tabs.

Comment (diaspora-code-review) [inline app/assets/javascripts/app/helpers/modal_helper.js]: Expected indentation of 8 space characters but found 6.

Comment (diaspora-code-review) [inline app/assets/javascripts/app/helpers/modal_helper.js]: Expected space(s) after "if".

Comment (diaspora-code-review) [inline app/assets/javascripts/app/helpers/modal_helper.js]: Missing space before opening brace.

File: app/assets/javascripts/app/helpers/modal_helper.js [hunk 8]
```diff
@@ -1,12 +1,34 @@
...
+          $modalTitle.text($modalTitle.data("original-title"));
```
Comment (diaspora-code-review) [inline app/assets/javascripts/app/helpers/modal_helper.js]: Expected indentation of 8 space characters but found 10.

File: app/assets/javascripts/app/helpers/modal_helper.js [hunk 9]
```diff
@@ -1,12 +1,34 @@
...
+        }
```
Comment (diaspora-code-review) [inline app/assets/javascripts/app/helpers/modal_helper.js]: Expected indentation of 6 space characters but found 8.

File: app/assets/javascripts/app/helpers/modal_helper.js [hunk 10]
```diff
@@ -1,12 +1,34 @@
...
+        var profileMention = $(id).find(".modal-body");
 
-    modalBody.load(url, function(){
-      $(id).find("#modalWaiter").remove();
-    });
+        var url = $(id).attr("href");
+        profileMention.load(url, function(){
```
Comment (diaspora-code-review) [inline app/assets/javascripts/app/helpers/modal_helper.js]: Missing space before opening brace.

File: app/assets/javascripts/app/views/hovercard_view.js [hunk 11]
```diff
@@ -5,15 +5,23 @@ app.views.Hovercard = app.views.Base.extend({
   id: 'hovercard_container',
 
   events: {
-    'mouseleave': '_mouseleaveHandler'
+    'mouseleave': '_mouseleaveHandler',
```
Comment (diaspora-code-review) [inline app/assets/javascripts/app/views/hovercard_view.js]: Strings must use doublequote.

File: app/assets/javascripts/app/views/hovercard_view.js [hunk 12]
```diff
@@ -5,15 +5,23 @@ app.views.Hovercard = app.views.Base.extend({
...
+    "click #mention_button": "showMentionModal",
+    "click #message_button": "showMessageModal",
```
Comment (diaspora-code-review) [inline app/assets/javascripts/app/views/hovercard_view.js]: Unexpected trailing comma.

File: app/assets/javascripts/app/views/hovercard_view.js [hunk 13]
```diff
@@ -5,15 +5,23 @@ app.views.Hovercard = app.views.Base.extend({
...
   },
 
   initialize: function() {
     this.render();
-
+
```
Comment (diaspora-code-review) [inline app/assets/javascripts/app/views/hovercard_view.js]: Trailing spaces not allowed.

File: app/assets/javascripts/app/views/hovercard_view.js [hunk 14]
```diff
@@ -5,15 +5,23 @@ app.views.Hovercard = app.views.Base.extend({
...
   },
 
   initialize: function() {
     this.render();
-
+    
     $(document)
       .on('mouseenter', '.hovercardable', _.bind(this._mouseenterHandler, this))
-      .on('mouseleave', '.hovercardable', _.bind(this._mouseleaveHandler, this));
+      .on('mouseleave', '.hovercardable', _.bind(this._mouseleaveHandler, this))
```
Comment (diaspora-code-review) [inline app/assets/javascripts/app/views/hovercard_view.js]: Strings must use doublequote.

File: app/assets/javascripts/app/views/hovercard_view.js [hunk 15]
```diff
@@ -5,15 +5,23 @@ app.views.Hovercard = app.views.Base.extend({
...
+      .on('click', '#mention_button', function() {
```
Comment (diaspora-code-review) [inline app/assets/javascripts/app/views/hovercard_view.js]: Strings must use doublequote.

File: app/assets/javascripts/app/views/hovercard_view.js [hunk 16]
```diff
@@ -5,15 +5,23 @@ app.views.Hovercard = app.views.Base.extend({
...
+        $('#hovercard_container').fadeOut('fast');
```
Comment (diaspora-code-review) [inline app/assets/javascripts/app/views/hovercard_view.js]: Strings must use doublequote.

File: app/assets/javascripts/app/views/hovercard_view.js [hunk 17]
```diff
@@ -5,15 +5,23 @@ app.views.Hovercard = app.views.Base.extend({
...
+      })
+      .on('click', '#message_button', function() {
```
Comment (diaspora-code-review) [inline app/assets/javascripts/app/views/hovercard_view.js]: Strings must use doublequote.

File: app/assets/javascripts/app/views/hovercard_view.js [hunk 18]
```diff
@@ -5,15 +5,23 @@ app.views.Hovercard = app.views.Base.extend({
...
+        $('#hovercard_container').fadeOut('fast');
```
Comment (diaspora-code-review) [inline app/assets/javascripts/app/views/hovercard_view.js]: Strings must use doublequote.

File: app/assets/javascripts/app/views/hovercard_view.js [hunk 19]
```diff
@@ -25,9 +33,12 @@ app.views.Hovercard = app.views.Base.extend({
     this.hashtags = this.$('.hashtags');
     this.person_link = this.$('a.person');
     this.person_handle = this.$('div.handle');
-    this.active = true;
+    this.person_mention_button = this.$('#mention_button');
```
Comment (diaspora-code-review) [inline app/assets/javascripts/app/views/hovercard_view.js]: Identifier 'person_mention_button' is not in camel case.

Comment (diaspora-code-review) [inline app/assets/javascripts/app/views/hovercard_view.js]: Strings must use doublequote.

File: app/assets/javascripts/app/views/hovercard_view.js [hunk 20]
```diff
@@ -25,9 +33,12 @@ app.views.Hovercard = app.views.Base.extend({
...
+    this.person_message_button = this.$('#message_button');
```
Comment (diaspora-code-review) [inline app/assets/javascripts/app/views/hovercard_view.js]: Identifier 'person_message_button' is not in camel case.

Comment (diaspora-code-review) [inline app/assets/javascripts/app/views/hovercard_view.js]: Strings must use doublequote.

File: app/assets/javascripts/app/views/hovercard_view.js [hunk 21]
```diff
@@ -25,9 +33,12 @@ app.views.Hovercard = app.views.Base.extend({
...
+    this.active = true;
```
Comment (diaspora-code-review) [inline app/assets/javascripts/app/views/hovercard_view.js]: Trailing spaces not allowed.

File: app/assets/javascripts/app/views/hovercard_view.js [hunk 22]
```diff
@@ -25,9 +33,12 @@ app.views.Hovercard = app.views.Base.extend({
     this.hashtags = this.$('.hashtags');
     this.person_link = this.$('a.person');
     this.person_handle = this.$('div.handle');
-    this.active = true;
+    this.person_mention_button = this.$('#mention_button');
+    this.person_message_button = this.$('#message_button');
+    this.active = true;  
   },
 
+
```
Comment (diaspora-code-review) [inline app/assets/javascripts/app/views/hovercard_view.js]: More than 1 blank line not allowed.

File: app/assets/javascripts/app/views/hovercard_view.js [hunk 23]
```diff
@@ -95,6 +106,7 @@ app.views.Hovercard = app.views.Base.extend({
   _populateHovercard: function() {
     var href = this.href();
     href += "/hovercard.json";
+    //console.log('calling ' + href);
```
Comment (diaspora-code-review) [inline app/assets/javascripts/app/views/hovercard_view.js]: Expected space or tab after '//' in comment.

File: app/assets/javascripts/app/views/hovercard_view.js [hunk 24]
```diff
@@ -113,12 +125,16 @@ app.views.Hovercard = app.views.Base.extend({
 
   _populateHovercardWith: function(person) {
     var self = this;
-
+
```
Comment (diaspora-code-review) [inline app/assets/javascripts/app/views/hovercard_view.js]: Trailing spaces not allowed.

File: app/assets/javascripts/app/views/hovercard_view.js [hunk 25]
```diff
@@ -113,12 +125,16 @@ app.views.Hovercard = app.views.Base.extend({
...
   _populateHovercardWith: function(person) {
     var self = this;
-
+    
     this.avatar.attr('src', person.avatar);
     this.avatarLink.attr("href", person.url);
     this.person_link.attr('href', person.url);
     this.person_link.text(person.name);
     this.person_handle.text(person.handle);
+    this.person_mention_button.attr('data-status-message-path', person.status_url);
```
Comment (diaspora-code-review) [inline app/assets/javascripts/app/views/hovercard_view.js]: Strings must use doublequote.

File: app/assets/javascripts/app/views/hovercard_view.js [hunk 26]
```diff
@@ -113,12 +125,16 @@ app.views.Hovercard = app.views.Base.extend({
...
+    this.person_mention_button.attr('data-title', person.title);
```
Comment (diaspora-code-review) [inline app/assets/javascripts/app/views/hovercard_view.js]: Strings must use doublequote.

File: app/assets/javascripts/app/views/hovercard_view.js [hunk 27]
```diff
@@ -113,12 +125,16 @@ app.views.Hovercard = app.views.Base.extend({
...
+    this.person_message_button.attr('data-conversation-path', person.message_url);
```
Comment (diaspora-code-review) [inline app/assets/javascripts/app/views/hovercard_view.js]: Strings must use doublequote.

File: db/migrate/20160709173109_change_collation_for_tag_names.acts_as_taggable_on_engine.rb [hunk 28]
```diff
@@ -0,0 +1,10 @@
+# This migration comes from acts_as_taggable_on_engine (originally 5)
+# This migration is added to circumvent issue #623 and have special characters
+# work properly
+class ChangeCollationForTagNames < ActiveRecord::Migration
+  def up
+    if ActsAsTaggableOn::Utils.using_mysql?
+      execute("ALTER TABLE tags MODIFY name varchar(255) CHARACTER SET utf8 COLLATE utf8_bin;")
+    end
+  end
+end
```
Comment (jhass) [inline db/migrate/20160709173109_change_collation_for_tag_names.acts_as_taggable_on_engine.rb]: You shouldn't have run the generator command for these migrations and not have committed them. Please Remove these and also any change to `db/schema.rb`.

**Comment by cmrd-senya** [comment]
I guess this one could conflict with #6864 which is complete, but yet not fully-reviewed. I'd love to avoid painful rebasing of my big PR on a conflicting base.

**Comment by jhass** [comment]
Please have a look at https://wiki.diasporafoundation.org/Styleguide#Automatic_local_review

File: db/migrate/20160709173109_change_collation_for_tag_names.acts_as_taggable_on_engine.rb (hunk 28 above)
Comment (angerella10) [inline db/migrate/20160709173109_change_collation_for_tag_names.acts_as_taggable_on_engine.rb]: Sure! Thanks for this.

**Comment by Flaburgan** [comment]
@angerella10 do you plan to work on this again?

**Comment by denschub** [comment]
Let's go ahead and close this PR. There is a lot of work left to do, and there wasn't too much engineering spent on here. If someone is interested in working on this, feel free to pick up this PR and base your own work on the work being done here, but for now, we have no intentions of continuing this work ourselves.

**Pull request closed without merging**